### PR TITLE
Adding `state-machine-cat`

### PR DESCRIPTION
**Pull request by tuanpham96**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tuanpham96** [comment]
@conda-forge/help-nodejs This is ready for review!

File: recipes/state-machine-cat/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2016-2022 Sander Verweij
```
Comment (sverweij) [inline recipes/state-machine-cat/LICENSE:3]:
```suggestion
Copyright (c) 2016-2023 Sander Verweij
```

(NTS: probably needs  to be adapted in the state-machine-cat license file  proper as  well.

File: recipes/state-machine-cat/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{% set name = "state-machine-cat" %}
+{% set version = "v10.1.10" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version|replace("-", "_")|replace("v", "") }}
+
+source:
+  url: https://github.com/sverweij/state-machine-cat/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 386d2578731f0899c540e5020b4627cc75af4e50fb61887505aac47bd5c9dea1
```
Comment (sverweij) [inline recipes/state-machine-cat/meta.yaml:10]: Is it possible to use the distribution at npmjs? That one is leaner to download as it only consists of the sources strictly necessary for running  state-machine-cat (~350kb) and doesn't contain the 3.5Mb of tests and 22Mb of website/ docs.

```suggestion
  url: https://registry.npmjs.org/state-machine-cat/-/state-machine-cat-{{ version|replace("v", "") }}.tgz
  sha256: 7348eaab734296dfa9aa046ad7ce20e1d64696c8e81c73019e08e3ad22b18e9a
```

:point_up: this sha256 is  b.t.w. the one for v11.0.0

File: recipes/state-machine-cat/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{% set name = "state-machine-cat" %}
+{% set version = "v10.1.10" %}
```
Comment (sverweij) [inline recipes/state-machine-cat/meta.yaml:2]:
```suggestion
{% set version = "v11.0.0" %}
```
v11.0.0 is the latest version (I noticed I haven't explicitly  documentet that  on  on GitHub yet - but it is the latest  there as well.

This is assuming the `node` version conda forge uses is >14

File: recipes/state-machine-cat/meta.yaml (hunk 2 above)
Comment (tuanpham96) [reply recipes/state-machine-cat/meta.yaml:10]: Not sure, but I can try that. 

Where's the sha256 for v11.0.0 documented by the way?

On the releases page, it says this, which is different from above.
> shasum of the package as published on npmjs: 1a3629148c765a814fd1a837df75b1ccfe21cb4d

Comment (sverweij) [reply recipes/state-machine-cat/meta.yaml:10]: npmjs seems to have only an md5 and a sha512 (you can check with `npm view state-machine-cat`) readily available. If conda-forge can handle a sha512 that would  be easiest.

I got the sha256 by downloading the archive and calculating  the sha256 from  it:

```sh
wget https://registry.npmjs.org/state-machine-cat/-/state-machine-cat-11.0.0.tgz
openssl sha256 state-machine-cat-11.0.0.tgz
```

Comment (tuanpham96) [reply recipes/state-machine-cat/meta.yaml]: Unfortunately `sha512` is not available yet (there's an open PR for that https://github.com/conda/conda-build/pull/4793). 

I'm using `sha256` for the time being. 

And the npm registry source works perfectly.

File: recipes/state-machine-cat/meta.yaml (hunk 3 above)
Comment (tuanpham96) [reply recipes/state-machine-cat/meta.yaml]: Yes, based on the ci logs, I updated the version and enforced `node >14` - everything looks good (based on the logs it uses `node 18.*`

**Comment by tuanpham96** [comment]
@conda-forge-admin, please ping conda-forge/help-nodejs

**Pull request closed without merging**